Panic at rend3::util::registry::archetypical::ArchetypicalRegistry<K,V,HandleType>::get_value_mut

```
thread 'main' panicked at 'no entry found for key', /home/john/.cargo/git/checkouts/rend3-e03f89403de3386a/a972b0f/rend3/src/util/registry/archetypical.rs:126:28
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
   2: core::panicking::panic_display
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:72:5
   3: core::panicking::panic_str
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:56:5
   4: core::option::expect_failed
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/option.rs:1874:5
   5: rend3::util::registry::archetypical::ArchetypicalRegistry<K,V,HandleType>::get_value_mut
   6: rend3::renderer::ready::ready
   7: rend3::renderer::Renderer::ready
   8: <sharpview::AppUi as rend3_framework::App>::handle_event
   9: rend3_framework::async_start::{{closure}}::{{closure}}
  10: winit::platform_impl::platform::x11::EventLoop<T>::run
  11: winit::platform_impl::platform::EventLoop<T>::run
  12: winit::event_loop::EventLoop<T>::run
  13: rend3_framework::winit_run
  14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  15: rend3_framework::start
  16: sharpview::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
Never saw this before.

My own code is 100% safe Rust. Recent changes made it more concurrent, and it may be hitting Rend3 with different timing.

Rend3 version info:
```
rend3 = { git = 'https://github.com/BVE-Reborn/rend3.git', rev = "a972b0ff0a98bb79b2b1c8812e3a823840a5e02e" }
rend3-routine = { git = 'https://github.com/BVE-Reborn/rend3.git', rev = "a972b0ff0a98bb79b2b1c8812e3a823840a5e02e" }
rend3-gltf = { git = 'https://github.com/BVE-Reborn/rend3.git', rev = "a972b0ff0a98bb79b2b1c8812e3a823840a5e02e" }

```
Yes, that's way out of date. I can't find "registry" or "archetyhpical" or "get_value_mut" in the current sources, so it appears that this code has been replaced. 

Waiting for the new re-factored release to be announced. I've built test programs with the top of "trunk" but have not modified my main program pending the completion of the current rework.

Trying to build branch "object-lifespan-logging". That uses wgpu 0.14 instead of the current 0.15,  an earlier version of egui, etc. Is that intentional? Trying to use that version of Rend (git clone from Github, git checkout object-lifespan-logging") I get errors building my own program as Rend3 compiles:

```
   Compiling rend3 v0.3.0 (/home/john/projects/rend3/rend3)
error[E0560]: struct `Limits` has no field named `max_bindings_per_bind_group`
  --> /home/john/projects/rend3/rend3/src/setup.rs:85:5
   |
85 |     max_bindings_per_bind_group: 640,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Limits` does not have this field
   |
   = note: available fields are: `max_texture_dimension_1d`, `max_texture_dimension_2d`, `max_texture_dimension_3d`, `max_texture_array_layers`, `max_bind_groups` ... and 23 others

error[E0560]: struct `Limits` has no field named `max_bindings_per_bind_group`
   --> /home/john/projects/rend3/rend3/src/setup.rs:117:5
    |
117 |     max_bindings_per_bind_group: 640,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Limits` does not have this field
    |
    = note: available fields are: `max_texture_dimension_1d`, `max_texture_dimension_2d`, `max_texture_dimension_3d`, `max_texture_array_layers`, `max_bind_groups` ... and 23 others

error[E0599]: no method named `unwrap` found for struct `QueueWriteBufferView<'_>` in the current scope
   --> /home/john/projects/rend3/rend3/src/managers/mesh.rs:130:18
    |
130 |                 .unwrap();
    |                  ^^^^^^ help: there is a method with a similar name: `swap`

error[E0599]: no method named `unwrap` found for struct `QueueWriteBufferView<'_>` in the current scope
   --> /home/john/projects/rend3/rend3/src/managers/mesh.rs:140:14
    |
140 |             .unwrap();
    |              ^^^^^^ help: there is a method with a similar name: `swap`

error[E0599]: no method named `unwrap` found for struct `QueueWriteBufferView<'_>` in the current scope
  --> /home/john/projects/rend3/rend3/src/util/buffer.rs:64:72
   |
64 |         let mut mapped = queue.write_buffer_with(&self.inner, 0, size).unwrap();
   |                                                                        ^^^^^^ help: there is a method with a similar name: `swap`

```

--- a/rend3-types/Cargo.toml
+++ b/rend3-types/Cargo.toml
@@ -13,7 +13,7 @@ rust-version = "1.65"
 
 [dependencies]
 bitflags = "1"
-bytemuck = "1"
+bytemuck = { version = "1", features = ["min_const_generics"] }
 encase = { version = "0.4", features = ["glam"] }
 glam = { version = "0.22", features = ["bytemuck"] }
 list-any = "0.2"

--- a/rend3-types/src/lib.rs
+++ b/rend3-types/src/lib.rs
@@ -16,6 +16,7 @@ use std::{
     sync::Arc,
 };
 
+use bytemuck::Zeroable;
 /// Reexport of the glam version rend3 is using.
 pub use glam;
 use glam::{Mat4, UVec2, Vec2, Vec3, Vec3A, Vec4};
@@ -995,7 +996,7 @@ pub enum SortingOrder {
 pub trait MaterialArray<T>: AsRef<[T]> {
     /// An array of the [u32; COUNT]. We need this internally
     /// for shader layout stuff.
-    type U32Array: encase::ShaderSize + encase::internal::WriteInto + Clone + Copy + Send + Sync + 'static;
+    type U32Array: encase::ShaderSize + encase::internal::WriteInto + Zeroable + Clone + Copy + Send + Sync + 'static;
     type IntoIter: Iterator<Item = T>;
     const COUNT: u32;
 

--- a/rend3/src/managers/object.rs
+++ b/rend3/src/managers/object.rs
@@ -1,5 +1,6 @@
 use std::{any::TypeId, ops::Range};
 
+use bytemuck::Zeroable;
 use encase::ShaderType;
 use glam::{Mat4, Vec3A};
 use list_any::VecAny;
@@ -30,6 +31,19 @@ pub struct ShaderObject<M: Material> {
     pub material_index: u32,
     pub vertex_attribute_start_offsets:
         <M::SupportedAttributeArrayType as MaterialArray<&'static VertexAttributeId>>::U32Array,
+}
+
+impl<M: Material> Default for ShaderObject<M> {
+    fn default() -> Self {
+        Self {
+            transform: Default::default(),
+            bounding_sphere: Default::default(),
+            first_index: Default::default(),
+            index_count: Default::default(),
+            material_index: Default::default(),
+            vertex_attribute_start_offsets: Zeroable::zeroed(),
+        }
+    }
 }
 
 // Manual impl so that M: !Copy
@@ -352,7 +366,7 @@ fn evaluate<M: Material>(
         .downcast_slice::<Option<InternalObject<M>>>()
         .unwrap();
 
-    archetype
-        .buffer
-        .apply(device, encoder, scatter, |idx| data_vec[idx].as_ref().unwrap().inner)
+    archetype.buffer.apply(device, encoder, scatter, |idx| {
+        data_vec[idx].as_ref().map(|o| o.inner).unwrap_or_default()
+    })
 }

--- a/rend3/src/util/frustum.rs
+++ b/rend3/src/util/frustum.rs
@@ -6,7 +6,7 @@ use encase::ShaderType;
 use glam::{Mat4, Vec3, Vec3A, Vec4Swizzles};
 
 /// Represents a point in space and a radius from that point.
-#[derive(Debug, Clone, Copy, ShaderType)]
+#[derive(Debug, Default, Clone, Copy, ShaderType)]
 pub struct BoundingSphere {
     pub center: Vec3,
     pub radius: f32,

